src/search-modal/SearchResult.jsx

```diff
@@ -1,37 +1,147 @@
+  *  'content.html_content'?: string,
+  *  'content.capa_content'?: string,
+  *  breadcrumbs: {display_name: string}[]
+  *  breadcrumbsNames: string[],
+  *  }>} CustomHit
+  */
+
+/**
+ * Custom Highlight component that uses the <b> tag for highlighting
+ * @type {React.FC<{
+ *   attribute: keyof CustomHit | string[],
+ *   hit: CustomHit,
+ *   separator?: string,
  * }>}
  */
-const SearchResult = ({ hit }) => (
-  <div className="my-2 pb-2 border-bottom">
-    <div className="hit-name small">
-      <strong><Highlight attribute="display_name" hit={hit} /></strong>{' '}
-      (<BlockTypeLabel type={hit.block_type} />)
-    </div>
-    <div className="hit-description x-small text-truncate">
-      <Highlight attribute="content.html_content" hit={hit} />
-      <Highlight attribute="content.capa_content" hit={hit} />
-    </div>
-    <div className="text-muted x-small">
-      {hit.breadcrumbs.map((bc, i) => (
-        // eslint-disable-next-line react/no-array-index-key
-        <span key={i}>{bc.display_name} {i !== hit.breadcrumbs.length - 1 ? '/' : ''} </span>
-      ))}
-    </div>
-  </div>
+const CustomHighlight = ({ attribute, hit, separator }) => (
+  <Highlight
+    attribute={attribute}
+    hit={hit}
+    separator={separator}
+    highlightedTagName="b"
+  />
 );
 
+const ItemIcon = {
+  vertical: Folder,
+  sequential: Folder,
+  chapter: Folder,
+  problem: Question,
+  video: Videocam,
+  html: TextFields,
+};
+
+/**
+ * Returns the URL for the context of the hit
+ * @param {CustomHit} hit
+ * @param {boolean} newWindow
+ * @param {string} libraryAuthoringMfeUrl
+ * @returns {string}
+ */
+const getContextUrl = (hit, newWindow, libraryAuthoringMfeUrl) => {
+  const { context_key: contextKey, usage_key: usageKey } = hit;
+  if (contextKey.startsWith('course-v1:')) {
+    if (newWindow) {
+      return `${getPath(getConfig().PUBLIC_PATH)}/course/${contextKey}?show=${encodeURIComponent(usageKey)}`;
+    }
+    return `/course/${contextKey}?show=${encodeURIComponent(usageKey)}`;
+  }
+  if (usageKey.includes('lb:')) {
+    return `${libraryAuthoringMfeUrl}library/${contextKey}`;
+  }
+  return '#';
+};
+
+/**
+ * A single search result (row), usually represents an XBlock/Component
+ * @type {React.FC<{ hit: CustomHit, closeSearch?: () => void}>}
+ */
+const SearchResult = ({ hit, closeSearch }) => {
+  const navigate = useNavigate();
+  const { libraryAuthoringMfeUrl } = useSelector(getStudioHomeData);
+
+  /**
+    * Navigates to the context of the hit
+    * @param {React.MouseEvent} e
+    * @param {boolean} newWindow
+    * @returns {void}
+    * */
+  const navigateToContext = (e, newWindow) => {
+    e.stopPropagation();
+    const url = getContextUrl(hit, newWindow, libraryAuthoringMfeUrl);
+    if (newWindow) {
+      window.open(url, '_blank');
+      return;
+    }
+
+    if (url.startsWith('http')) {
+      window.location.href = url;
+      return;
+    }
+
+    navigate(url);
+    closeSearch?.();
```

Comment:
I had to pass the `closeSearch` function down the path `SearchModal > SearchEndpointLoader > SearchUI > SearchResult` to close the modal. Maybe we should add a context to handle this state?

Reply:
We don't have this problem in the other modals because we don't have a deep inner component interacting with it.
The current approach is the simple-but-not-so-beauty way. We could hear more from cc/upstream review if we can improve it without adding complexity.

Reply:
I added it to a context in my PR when I merged it with this one - see [here](https://github.com/open-craft/frontend-app-course-authoring/compare/taxonomy-sandbox-20240412...open-craft:frontend-app-course-authoring:taxonomy-sandbox-20240412-b).